fix: show k8s schema validation error messages on each line separate

amustaque97 opened the pull request:
Fixes #350 

**Changes Proposed in PR:**
- In order to print structured error output I have added a new line in `func (e *InvalidK8sSchemaError) Error() string` method.
- Inside `ValidateResource` method replacing `-` with newline.

**Testing Instructions**
1. Create a yaml file with below content.
```yaml
apiVersion: v1
kind: pod
metaData:
  name: static-web
  labels:
    role: myrole
spec:
  containers:
    - name: web
      mage: nginx
      ports:
        - name: web
          containerPort: 80
          protocol: TCP

```
2. checkout to this branch
3.  build project
4. now run `./datree test <test-config.yaml`
5. Verify each error is coming on separate line.

**Screenshots**
<img width="782" alt="Screenshot 2022-01-03 at 2 26 42 AM" src="https://user-images.githubusercontent.com/21127788/147889424-d4df5e27-9721-4be3-b9d3-ec3cf7d02571.png">

cc @myishay @eyarz

amustaque97 committed:
fix: show structure error message on each line separate

myishay commented on bl/validation/k8sValidator.go:
```diff
@@ -124,7 +125,8 @@ func (val *K8sValidator) validateResource(filepath string) (bool, []error, error
 		if res.Status == kubeconformValidator.Invalid || res.Status == kubeconformValidator.Error {
 			isValid = false
-			validationErrors = append(validationErrors, &InvalidK8sSchemaError{ErrorMessage: res.Err.Error()})
+			errorMessage := strings.ReplaceAll(res.Err.Error(), "-", "\n   ")
```
I think it would be better to split this string into different "messages" and add each message as an element to the `validationErrors` array

myishay reviewed (changes requested):
@amustaque97 Thank you for your contribution!

I gave you one comment I think should be changed.
With my suggested changes in the comment, you will probably see an outcome that prints `❌  k8s schema validation error:` several times (for each validation error). If you have an idea for printing this line only once (like you suggested in the above-mentioned image) it could be great, otherwise I think it will be ok printing it for each validation error.

amustaque97 committed:
fix: print indiviual error on seperate line with prefix text

amustaque97 commented:
I have addressed review comment. Final output look like this:

<img width="857" alt="Screenshot 2022-01-03 at 6 48 04 PM" src="https://user-images.githubusercontent.com/21127788/147935025-8a7e6d97-3a91-46f0-9ae6-7de886baf51c.png">

myishay commented:
@amustaque97 LGTM!
I'll update your branch with the latest changes and merge.

Commit:
Merge branch 'main' into fix/show-separate-errors
